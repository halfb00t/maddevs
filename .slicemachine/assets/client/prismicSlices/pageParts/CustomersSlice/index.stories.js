import MyComponent from '../../../../../../client/prismicSlices/pageParts/CustomersSlice';
import SliceZone from 'vue-slicezone'

export default {
  title: 'client/prismicSlices/pageParts/CustomersSlice'
}


export const _DefaultSlice = () => ({
  components: {
    MyComponent,
    SliceZone
  },
  methods: {
    resolve() {
      return MyComponent
    }
  },
  data() {
    return {
      mock: {"variation":"default-slice","name":"Default slice","slice_type":"customers_slice","items":[],"primary":{"animation":"zoom-in-down","background":"white"},"id":"_DefaultSlice"}
    }
  },
  template: '<SliceZone :slices="[mock]" :resolver="resolve" />'
})
_DefaultSlice.storyName = 'Default slice'
