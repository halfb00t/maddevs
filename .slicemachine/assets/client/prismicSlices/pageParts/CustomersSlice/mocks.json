[
  {
    "variation": "default-slice",
    "name": "Default slice",
    "slice_type": "customers_slice",
    "items": [],
    "primary": {
      "animation": "zoom-in-down",
      "background": "white"
    }
  }
]