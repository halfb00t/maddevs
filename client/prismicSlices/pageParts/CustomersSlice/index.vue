<template>
  <section
    class="customer-logo-slice"
    :style="{ 'background-color': sliceBackground }"
  >
    <div
      class="container"
      :data-aos="animation"
    >
      <div
        data-testid="customer-logo-slice-container"
        :data-aos="animation"
        class="customer-logo-slice_list"
      >
        <div
          v-for="(customer, i) of sortedCursomersByPriority"
          :key="`customer-logo-slice-${i}`"
          class="customer-logo-slice_logo"
          data-testid="customer-logo-slice-item"
        >
          <img
            v-lazy-load
            :data-src="customer.gallery_image.url"
            :alt="customer.gallery_image.alt || 'Image'"
          >
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import { mapActions, mapGetters } from 'vuex'
import animateOnScrollMixin from '@/mixins/animateOnScrollMixin'
import setSliceBackground from '@/helpers/setSliceBackground'

export default {
  name: 'CustomersSlice',

  mixins: [animateOnScrollMixin({
    offset: 200,
    delay: 50,
    anchorPlacement: 'top-center',
    duration: 1000,
    once: true,
  })],

  props: {
    slice: {
      type: Object,
      required: true,
      default() {
        return {}
      },
    },
  },

  data() {
    return {
      animation: this.slice?.primary?.animation,
      sliceBackground: setSliceBackground(this.slice?.primary?.background),
    }
  },

  computed: {
    ...mapGetters(['sortedCursomersByPriority']),
  },

  created() {
    this.getCustomers()
  },

  methods: {
    ...mapActions(['getCustomers']),
  },
}
</script>

<style lang="scss" scoped>
.customer-logo-slice {
  .container {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &_list {
    width: 100%;
    display: flex;
    gap: 20px;
    justify-content: center;
    flex-wrap: wrap;

    @media only screen and (max-width: 1024px) {
      gap: 14px;
    }
  }

  &_logo {
    width: 16%;
    max-width: 190px;
    min-height: 111px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 24px 8px;
    box-sizing: border-box;

    img {
      width: 143px;
      height: 64px;
      display: block;
    }

    @media only screen and (max-width: 1160px) {
      width: 19%;
    }

    @media only screen and (max-width: 1024px) {
      width: 15%;
      min-height: 81px;

      img {
        width: 75px;
        height: 33px;
      }
    }

    @media only screen and (max-width: 834px) {
      width: 18%;
    }

    @media only screen and (max-width: 640px) {
      width: 22.99%;
    }

    @media only screen and (max-width: 520px) {
      width: 29%;
    }

    @media only screen and (max-width: 360px) {
      width: 45%;
    }
  }

  &_link {
    display: block;
    width: 100%;
    margin-top: 50px;

    /deep/ .ui-button {
      padding: 25px;
      line-height: 20px;

      &:hover {
        background-color: lighten($bgcolor--red, 10%) !important;
      }
    }

    @media screen and (max-width: 1024px) {
      margin-top: 32px;
      /deep/ .ui-button {
        padding: 14px 25px;
      }
    }
  }
}
</style>
