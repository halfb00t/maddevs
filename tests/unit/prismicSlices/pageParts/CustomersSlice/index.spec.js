import { createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import { render, screen } from '@testing-library/vue'
import CustomersSlice from '@/prismicSlices/pageParts/CustomersSlice'

const localVue = createLocalVue()
localVue.use(Vuex)

const actions = {
  getCustomers: () => {},
}

const getters = {
  sortedCursomersByPriority: () => ([
    {
      gallery_image: {
        url: 'https://superpupertest.cdn.prismic.io/superpupertest/8829fbb2-90af-4ded-913c-d503046e5ae7_clutch-black.svg',
        alt: 'Test',
      },
    },
  ]),
}

const store = new Vuex.Store({
  actions,
  getters,
})

const directives = {
  'lazy-load': () => {},
}

const apiData = {
  background: 'black',
  animation: 'fade-up',
}

const getProps = params => ({
  slice: {
    primary: {
      ...params,
    },
  },
})

describe('Custom logo slice', () => {
  it('should render correctly with data', () => {
    const { container } = render(CustomersSlice, {
      directives,
      store,
      props: getProps(apiData),
    })
    expect(screen.getAllByTestId('customer-logo-slice-item')).not.toBe(null)
    expect(container).toMatchSnapshot()
  })
})

describe('Custom logo slice data-aos animation attribute', () => {
  it(`should the data-aos to be '${apiData.animation}'`, () => {
    render(CustomersSlice, {
      directives,
      store,
      props: getProps(apiData),
    })

    expect(screen.getByTestId('customer-logo-slice-container').getAttribute('data-aos')).toBe(apiData.animation)
  })

  it('should the data-aos to be null', () => {
    render(CustomersSlice, {
      directives,
      store,
      props: getProps({}),
    })
    expect(screen.getByTestId('customer-logo-slice-container').getAttribute('data-aos')).toBeNull()
  })
})
