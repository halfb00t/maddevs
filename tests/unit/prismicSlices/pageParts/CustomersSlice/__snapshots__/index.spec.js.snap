// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom logo slice should render correctly with data 1`] = `
<div>
  <section
    class="customer-logo-slice"
    style="background-color: rgb(17, 18, 19);"
  >
    <div
      class="container aos-init aos-animate"
      data-aos="fade-up"
    >
      <div
        class="customer-logo-slice_list aos-init aos-animate"
        data-aos="fade-up"
        data-testid="customer-logo-slice-container"
      >
        <div
          class="customer-logo-slice_logo"
          data-testid="customer-logo-slice-item"
        >
          <img
            alt="Test"
            data-src="https://superpupertest.cdn.prismic.io/superpupertest/8829fbb2-90af-4ded-913c-d503046e5ae7_clutch-black.svg"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
